.github/workflows/code-checking.yml

```diff
@@ -0,0 +1,24 @@
+name: Code Checks
+
+on:
+  pull_request:
+
+jobs:
+  code-checking:
+    name: Run Prettier and ESLint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+      - name: Install dependencies
+        run: npm install
+      - name: Compile Typescript to Javascript
```

Comment:
I don't think this step is needed here as prettier and linter are run agains the ts files